Implement nameOverrides and fullnameOverrides

ojhughes opened the pull request:
Don't use .Release.Name for naming kubernetes objects as it causes issues when the chart is used as a dependency

#38
#76

ojhughes committed:
Implement nameOverrides and fullnameOverrides

Don't use .Release.Name for naming kubernetes objects as it causes issues when the chart is used as a dependency

#38
#76

harshitsinghvi22 commented on neo4j/templates/_helpers.tpl:
```diff
@@ -1,4 +1,39 @@
+{{- define "neo4j.chart" -}}
+    {{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
```
are neo4j.Chart and neo4j.defaultName being used anywhere in the helm chart...i do not see any reference in the PR

ojhughes replied:
removed now

ojhughes committed:
Add unit tests for fullname override
Remove unused `defaultName` and `chartName` includes from _helpers.tpl
Remove commented out includes from _helpers.tpl

harshitsinghvi22 reviewed (approved):
Well done !!
